common/config/fx.go

```diff
@@ -49,7 +50,7 @@ func provideMembershipConfig(cfg *Config) *Membership {
 	return &cfg.Global.Membership
 }
 
-func provideServicePortMap(cfg *Config) ServicePortMap {
+func provideServicePortMap(cfg *Config, _ *net.Listener) ServicePortMap {
```

Comment:
I don't think you need this extra param. It's also what's causing the tests to fail

Reply:
How do you think we can enforce this function to be called only after the listener is created? It could read the port before it is set to its actual value.

I checked the documentation of `fx` and it doesn't seem to guarantee an order for constructors that have no dependencies

Reply:
Oh, I see. I don't know if it's going to be that simple. I think the best approach would be to probably refactor this so that we aren't reading raw port values. We don't want to do any side-effects during graph building time, only during the startup lifecycle, so we can pass around functions to read the ports, and then only actually read them in the start methods of objects. It will probably be a much larger change